src/lib/spinel/multi_frame_buffer.hpp

```diff
@@ -371,7 +371,7 @@ template <uint16_t kSize> class MultiFrameBuffer : public FrameWritePointer
 
         aFrame = (aFrame == nullptr) ? mBuffer : aFrame + aLength;
 
-        if (HasSavedFrame() && (aFrame != mWriteFrameStart))
+        if (HasSavedFrame() && (aFrame != mWriteFrameStart) && ((aFrame + kHeaderSize) < GetArrayEnd(mBuffer)))
```

Comment:
@zhanglongxia can you please share how we may get into bad state where we would need the `((aFrame + kHeaderSize) < GetArrayEnd(mBuffer))`.

I tried to check the code, but cannot see how this can happen. 

In `SaveFrame()` we have a check which makes sure we have bytes left for the next frame header, otherwise we fail (not save the frame and not change `mWriteFrameStart`).
```c++
    otError SaveFrame(void)
    {
        otError error = OT_ERROR_NONE;

        // If the next header will overflow the buffer, we can't save the frame.
        if (!CanWrite(kHeaderSize))
        {
            error = OT_ERROR_NO_BUFS;
        }
        else
        {
            LittleEndian::WriteUint16(GetSkipLength() + GetLength(), mWriteFrameStart + kHeaderTotalLengthOffset);
            mWriteFrameStart = mWritePointer;
            IgnoreError(SetSkipLength(0));
        }

        return error;
    }
```

So we should always have at least `kHeaderSize` remaining in buffer after `mWriteFrameStart`?

Reply:
@zhanglongxia I think this was addressed in 
- https://github.com/openthread/openthread/pull/9244.